kommunicate/src/main/java/io/kommunicate/KmConversationHelper.java

```diff
@@ -385,70 +388,53 @@ public void onReceive(KMUser user, ResultReceiver resultReceiver) {
             }
             return;
         }
 
         if (conversationBuilder.getContext() == null) {
             if (callback != null) {
-                callback.onFailure("Context cannot be null");
+                callback.onFailure(Utils.getString(conversationBuilder.getContext(), R.string.km_context_cannot_be_null));
             }
             return;
         }
 
-        if (Kommunicate.isLoggedIn(conversationBuilder.getContext())) {
-            try {
-                startConversation(conversationBuilder,
-                        getStartConversationHandler(conversationBuilder.isSkipConversationList(), false, null, callback));
-            } catch (KmException e) {
-                if (callback != null) {
-                    callback.onFailure(e);
-                }
-            }
-        } else {
-            if (!TextUtils.isEmpty(conversationBuilder.getAppId())) {
-                Kommunicate.init(conversationBuilder.getContext(), conversationBuilder.getAppId());
-            } else {
-                if (TextUtils.isEmpty(Applozic.getInstance(conversationBuilder.getContext()).getApplicationKey())) {
-                    if (callback != null) {
-                        callback.onFailure("Application ID cannot be empty");
-                    }
-                }
+        if (!(conversationBuilder.getContext() instanceof Activity)) {
+            if (callback != null) {
+                callback.onFailure(Utils.getString(conversationBuilder.getContext(), R.string.km_method_needs_activity_context));
             }
-            if (conversationBuilder.isWithPreChat()) {
-                try {
-                    Kommunicate.launchPrechatWithResult(conversationBuilder.getContext(), new KmPrechatCallback() {
-                        @Override
-                        public void onReceive(KMUser user, ResultReceiver resultReceiver) {
-                            Kommunicate.login(conversationBuilder.getContext(), user, getLoginHandler(conversationBuilder, getStartConversationHandler(conversationBuilder.isSkipConversationList(), false, resultReceiver, callback), callback));
+            return;
+        }
+
+        ApplozicConversation.getLatestMessageList(conversationBuilder.getContext(), false, new MessageListHandler() {
+            @Override
+            public void onResult(List<Message> messageList, ApplozicException e) {
+                if (e == null) {
+                    if (messageList != null) {
+                        if (messageList.size() == 0) {
+                            conversationBuilder.setSkipConversationList(false);
+                            conversationBuilder.launchConversation(callback);
+                        } else if (messageList.size() == 1) {
+                            openParticularConversation(conversationBuilder.getContext(), false, messageList.get(0).getGroupId(), callback);
+                        } else if (messageList.size() > 1) {
```

Comment:
condition inside if is not required here

Reply:
Done